### PR TITLE
Added `initiallySelectedNamspaces`

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

Added support for providing an `initiallySelectedNamspaces` which accepts an array of Namespaces to have selected by default. This can be helpful if you are using Namespaces but don't want some of them to show automatically as they might cause noise.

In this screenshot you can see in the bottom left we had set "Default" for `initiallySelectedNamspaces`:

Screenshot 2024-02-10 at 3 49 11 PM

This may be impacted by https://github.com/backstage/backstage/pull/22858. If this PR gets in before this one then this will get refactored and be included in that documentation.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -161,7 +161,10 @@ const routes = (
     <Route path="/home" element={<HomepageCompositionRoot />}>
       {homePage}
     </Route>
-    <Route path="/catalog" element={<CatalogIndexPage />} />
+    <Route
+      path="/catalog"
+      element={<CatalogIndexPage initiallySelectedNamespaces={['default']} />}
```
Comment (vinzscam) [inline packages/app/src/App.tsx:166]: perhaps we can leave this as it was and remove `different-namespace.yaml`?

Comment (awanlin) [reply packages/app/src/App.tsx:166]: Yes, this was not meant to be checked in, was part of my manual testing. I would like to keep `different-namespace.yaml` as it's handy to have an example in a different namespace as the current ones are all in the default

Comment (awanlin) [reply packages/app/src/App.tsx]: These changes have been made, I did remove `different-namespace.yaml` as well, it's easy to recreate if needed 👍

**Comment by awanlin** [comment]
@vinzscam any chance you can give this one another look, please?

File: .changeset/tricky-flies-destroy.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog-react': patch
+'@backstage/plugin-catalog': patch
+---
+
+Added support for providing an `initiallySelectedNamspaces` which accepts an array of Namespaces to have selected by default
```
Comment (vinzscam) [inline .changeset/tricky-flies-destroy.md:6]:
```suggestion
Added support for providing an `initiallySelectedNamespaces` which accepts an array of Namespaces to have selected by default
```

**Review by vinzscam — APPROVED** [review]

File: .changeset/tricky-flies-destroy.md (hunk 2 above)
Comment (awanlin) [reply .changeset/tricky-flies-destroy.md]: Oh, good catch! Fixed 👍

**Comment by awanlin** [comment]
Hi @Edje-C, any chance you can give this a look, please?

**Review by Edje-C — APPROVED** [review]
Looks good 👍🏾

**Comment by awanlin** [comment]
Thanks for the review @vinzscam! 👍 

Thanks for the review and merge @Edje-C! 🚀

**Pull request merged**